llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -10337,9 +10356,12 @@ PreservedAnalyses LoopVectorizePass::run(Function &F,
 
   auto &MAMProxy = AM.getResult<ModuleAnalysisManagerFunctionProxy>(F);
   PSI = MAMProxy.getCachedResult<ProfileSummaryAnalysis>(*F.getParent());
-  BFI = nullptr;
-  if (PSI && PSI->hasProfileSummary())
-    BFI = &AM.getResult<BlockFrequencyAnalysis>(F);
+  BlockFrequencyInfo *BFI = nullptr;
+  GetBFI = [&AM, &F, &BFI]() -> BlockFrequencyInfo & {
+    if (!BFI)
+      BFI = &AM.getResult<BlockFrequencyAnalysis>(F);
+    return *BFI;
```

Comment:
I think the main saving would come from avoiding the indirect call to the lambda in many cases.

Reply:
IIRC in LoopVectorize we only call it once to determine if optimizing for size, so there calling the lambda shouldn't make a difference